test/unit/specs/components/transactions/LiStakeTransaction.spec.js

```diff
@@ -0,0 +1,116 @@
+import { shallowMount } from "@vue/test-utils"
+import LiStakeTransaction from "transactions/LiStakeTransaction"
+import { stakingTxs } from "../../store/json/txs"
+import { state } from "renderer/connectors/lcdClientMock.js"
+
+describe(`LiStakeTransaction`, () => {
+  let wrapper
+  const validators = state.candidates
+  const propsData = {
+    transaction: stakingTxs[0],
+    txType: `cosmos-sdk/MsgCreateValidator`,
+    validators,
+    url: `/validator`,
+    bondingDenom: `stake`
+  }
+  it(`create validator`, () => {
+    wrapper = shallowMount(LiStakeTransaction, {
+      propsData,
+      stubs: [`router-link`]
+    })
+    expect(wrapper.vm.$el).toMatchSnapshot()
+  })
+
+  it(`edit validator`, async () => {
+    propsData.transaction = stakingTxs[1]
+    propsData.txType = `cosmos-sdk/MsgEditValidator`
+
+    wrapper = shallowMount(LiStakeTransaction, {
+      propsData,
+      stubs: [`router-link`]
+    })
+    expect(wrapper.vm.$el).toMatchSnapshot()
+  })
+
+  describe(`delegations`, () => {
+
+    it(`should show delegations`, () => {
+      propsData.transaction = stakingTxs[2]
+      propsData.txType = `cosmos-sdk/MsgDelegate`
+
+      wrapper = shallowMount(LiStakeTransaction, {
+        propsData,
+        stubs: [`router-link`]
+      })
+
+      expect(wrapper.vm.$el).toMatchSnapshot()
+    })
+
+    it(`should show the validator address if no validator with that address in store`, () => {
+      wrapper.setProps({
+        validators: []
+      })
+      expect(
+        wrapper
+          .text()
+          .indexOf(
+            `To cosmosvaladdr15ky9du8a2wlstz6fpx3p4mqpjyrm5ctqzh8yqw`
+          ) === -1
+      ).toBe(true)
+    })
+  })
+
+  describe(`unbonding delegations`, () => {
+    it(`should show unbondings and calculate tokens from shares`, () => {
```

Comment:
we should test `calculateTokens` on  `scripts/common` instead. It's not tested and a lot of the helper functions from the scripts aren't too.

Reply:
I created an issue #2177 to address that

Reply:
Also. I meant having an expect that checks if the value is in the UI instead of just storing a snapshot. I forget that sometimes too, but I think it would be beneficial when reading the tests. What do you think?